From e1418c08508aee8861aa6cd1bd7e4bc0be5259f7 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 21 Jul 2022 20:10:59 +0300
Subject: [PATCH] [SW-94864] rom: export image validation apis

Export image validation apis for re-use in ppboot.

Renamed the api for better readability.

Change-Id: Ica7d88f77b817b096b46926c008632002ecc7059
---

diff --git a/arch/gaudi3/arc/arcboot_rom.txt b/arch/gaudi3/arc/arcboot_rom.txt
index a28314f..52ebe4b 100644
--- a/arch/gaudi3/arc/arcboot_rom.txt
+++ b/arch/gaudi3/arc/arcboot_rom.txt
@@ -13,3 +13,5 @@
 boot_info_set_no_merge
 boot_info_get_no_merge
 hl_get_boot_info_ver_rom
+secureboot_verify_image_signature
+secureboot_verify_image_hash384
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 817bf28..dacb1f5 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -867,7 +867,7 @@
 
 	img_plus_key_len = boot_ptr.boot_ptrs_img_sz - boot_ptr.sig_sz - crc_bytes;
 
-	ret = secureboot_verify_image_v2((uint8_t *)boot_ptr_addr, img_plus_key_len,
+	ret = secureboot_verify_image_signature((uint8_t *)boot_ptr_addr, img_plus_key_len,
 			(uint8_t *)(boot_ptr_addr + img_plus_key_len - boot_ptr.pub_key_sz), boot_ptr.pub_key_sz,
 			(uint8_t *)(boot_ptr_addr + img_plus_key_len), boot_ptr.sig_sz,
 			boot_ptr.sec_ver, boot_ptr.key_idx);
diff --git a/arch/gaudi3/arc/dev_security.h b/arch/gaudi3/arc/dev_security.h
index e9e08e5..abe5648 100644
--- a/arch/gaudi3/arc/dev_security.h
+++ b/arch/gaudi3/arc/dev_security.h
@@ -71,7 +71,7 @@
 	HL_SEC_HBW_RR_SHORT_7 = 7,
 };
 
-int secureboot_verify_image_v2(const uint8_t *image, uint32_t image_len, uint8_t *key,
+int secureboot_verify_image_signature(const uint8_t *image, uint32_t image_len, uint8_t *key,
 				uint32_t key_len, uint8_t *sig, uint32_t sig_len,
 				uint8_t sec_ver, uint8_t key_index);
 
diff --git a/arch/gaudi3/arc/secureboot.c b/arch/gaudi3/arc/secureboot.c
index 5f76724..a968585 100644
--- a/arch/gaudi3/arc/secureboot.c
+++ b/arch/gaudi3/arc/secureboot.c
@@ -48,7 +48,7 @@
  * SECUREBOOT_VERIFY_KEY_FAILED : Key hash verification failed
  * SECUREBOOT_VERIFY_FAILED     : Signature validation failed
  */
-int secureboot_verify_image_v2(const uint8_t *image, uint32_t image_len, uint8_t *key,
+int secureboot_verify_image_signature(const uint8_t *image, uint32_t image_len, uint8_t *key,
 				uint32_t key_len, uint8_t *sig, uint32_t sig_len,
 				uint8_t sec_ver, uint8_t key_index)
 {
